### DISCUSSION TITLE
Visualization plugin as dependency

### DISCUSSION THREAD

**Discussion by naringleb** [gephi/gephi#2542] [Q&A]
I am using Gephi for molecular networks in my Java Swing application. At the beginnning the Toolkit worked just fine, but now I do not get the PreviewMouseListener working (I copied the code from the Toolkit Demo) and it seems like I rather need Gephi's visualization panel instead of the preview for some more interactive options. 
 Unfortunately, the gradle import via `api "org.gephi:visualization:0.9.3"` in the build.gradle does not work. Is there a way to use the visualization plugin although I am not using any netbeans framework (I assumed it is not working because it is a netbeans module)?
.`

**Comment by mbastian** [comment]
Hi @nagorczak, the Toolkit by default does NOT include the OpenGL visualization but only the Preview one. If that's enough for you, I suggest to rather try to debug what's going on with this `PreviwMouseListener` instead of trying to depend on the OpenGL engine. What do you think?

**Reply by naringleb** [reply]
I tried to debug this but the `PreviewMouseListener` is not added to the Renderer automatically and I don't know how to manually add it (I checked that the required annotation is added in the PreviewMouseListener class). I now solved it by implementing a normal Swing MouseListener for the PreviewPanel. But thank you for responding!
